### ISSUE TITLE
Allow for cargo dylib compilation [phab:D938]

### ISSUE THREAD

**Issue by kythe** [google/kythe#1258]
_Created by **djrenren** at 2016-06-21 20:11:42_: 

We need the ability to compile dylibs as well as static libs (rlibs) so we can build the indexer as a compiler plugin (which must be a dylib).

**Comment by kythe** [comment]
_**shahms** commented inline (context lost during migration) at 2016-06-21 20:39:57_:

This extension is target platform dependent; `.so` isn't reliable.

**Comment by kythe** [comment]
_**shahms** commented inline (context lost during migration) at 2016-06-21 21:06:27_:

And, unfortunately, I'm not sure how to do this in a platform independent fashion short of hard-coding the expected extension for each platform (Linux, macOS, Windows)

**Comment by kythe** [comment]
_**shahms** wrote at 2016-06-21 21:46:20_:

I should be able to give you SSH access to kythe-macmini.sea.corp.google.com if you want to test on macOS.

**Comment by kythe** [comment]
_**djrenren** wrote at 2016-06-22 00:14:17_:

  - Implement cross-platform rust shared libs

**Comment by kythe** [comment]
_**djrenren** wrote at 2016-06-22 00:16:16_:

Added a tag called "arc-ignore" (completely open to changing this) that causes a rule to not get run at `arc diff` time. The reason for this is that these rules are platform dependent and the appropriate one will be selected by the filegroup rule.

**Comment by kythe** [comment]
_**zarko** wrote at 2016-06-22 04:30:40_:

On macOSx without Rust installed, `bazel test //...` gives:
```
FAIL: //kythe/rust/indexer:test_it_works (see /private/var/tmp/_bazel_zarko/2be6994bf77f75bac8f1433eb6e9dc33/execroot/kythe-ci/bazel-out/local_darwin-fastbuild/testlogs/kythe/rust/indexer/test_it_works/test.log).
```
The test passed after installing Rust.

On Linux without Rust installed, I got the same sort of error:
```
FAIL: //kythe/rust/indexer:test_it_works (see /usr/local/google/home/zarko/.cache/bazel/_bazel_zarko/b89de19a538f272d8732af51d9c5c7fb/kythe-trunk/bazel-out/local_linux-fastbuild/testlogs/kythe/rust/indexer/test_it_works/test.log).
```

The contents of the file is
```
\nERROR: No Rust Installation Found\n
```

With Rust installed, things appeared to work.

Are these expected [to go away after this diff is submitted]?

**Comment by kythe** [comment]
_**djrenren** wrote at 2016-06-22 05:24:03_:

zarko whoops looks like tests aren't tagged as manual! Must've lost that somewhere in refactoring. Will make the fix tomorrow.

**Comment by kythe** [comment]
_**shahms** accepted this revision at 2016-06-22 17:21:08_

**Comment by kythe** [comment]
_**djrenren** closed via commit d90e65d899f03e0a8ee412600c602800786a2c83 at 2016-06-22 17:34:24_
